controllers/machine_controller_phases.go

```diff
@@ -254,22 +254,23 @@ func (r *MachineReconciler) reconcileInfrastructure(ctx context.Context, cluster
 	if infraReconcileResult.Paused {
 		return ctrl.Result{}, nil
 	}
-	infraConfig := infraReconcileResult.Result
 
-	if !infraConfig.GetDeletionTimestamp().IsZero() {
+	r.infraReconcileResult = infraReconcileResult.Result
```

Comment:
I think you'll end up reverting all the changes in this file (re race condition)